WCFDataService/Client/System/Data/Services/Client/ClientEdmModel.cs

```diff
@@ -495,12 +499,46 @@ private EdmTypeCacheValue GetOrCreateEdmTypeInternal(IEdmStructuredType edmBaseT
         }
 
         /// <summary>
+        /// Tries to find the backing field for property on type.
+        /// </summary>
+        /// <param name="property">The property to find the backing field for.</param>
+        /// <returns>The property backing field or null.</returns>
+        private static FieldInfo FindBackingField(PropertyInfo property, ProxyBackingFieldNamingConvention proxyBackingFieldNamingConvention)
+        {
+            FieldInfo backingField = null;
+            Type propertyType = property.PropertyType;
+            // We only do this for "generic collections of entities" OR "complex properties"
+            if (PlatformHelper.IsGenericType(propertyType) && ClientTypeUtil.TypeOrElementTypeIsEntity(propertyType) && TypeSystem.FindIEnumerable(propertyType) != null ||
```

Comment:
Shoud be "if (PlatformHelper.IsGenericType(propertyType) && ClientTypeUtil.TypeOrElementTypeIsEntityOrComplex(propertyType) && TypeSystem.FindIEnumerable(propertyType) != null)" ?
Current condition does not cover collection of complex property which type is ObservableCollection<> is our auto-generated proxy file.
And in auto-generated proxy file, seems single complex type is already lazy initialized, for example:
_get
            {
                if (((this._PrimaryContactInfo == null)
                            && (this._PrimaryContactInfoInitialized != true)))
                {
                    this._PrimaryContactInfo = new ContactDetails();
                    this._PrimaryContactInfoInitialized = true;
                }
                return this._PrimaryContactInfo;
            }_